### PR TITLE
[MESS] spc1000 added the centronics printer port

### PR DESCRIPTION
**Pull request by meesokim**
The printer port is shared by cass output port (0x6000).
I add the port strobe signal handling code to support the printer.
The feature has been confirmed.

### CODE REVIEW

File: src/mess/drivers/spc1000.c [hunk 1]
```diff
@@ -477,6 +491,12 @@ static MACHINE_CONFIG_START( spc1000, spc1000_state )
 	MCFG_DEVICE_ADD("ext1", SPC1000_EXP_SLOT, 0)
 	MCFG_DEVICE_SLOT_INTERFACE(spc1000_exp, NULL, false)
 
+	MCFG_CENTRONICS_ADD("centronics", centronics_devices, "printer")
+	MCFG_CENTRONICS_ACK_HANDLER(DEVWRITELINE("cent_status_in", input_buffer_device, write_bit4))
```
Comment (startaq) [inline src/mess/drivers/spc1000.c]: ACK doesn't seem to be used anywhere.

File: src/mess/drivers/spc1000.c [hunk 2]
```diff
@@ -217,10 +224,16 @@ READ8_MEMBER(spc1000_state::iplk_r)
 WRITE8_MEMBER( spc1000_state::cass_w )
 {
 	bool m = BIT(data, 1) ? true : false;
+	bool s = BIT(data, 2) ? true : false;
 	m_cass->output(BIT(data, 0) ? -1.0 : 1.0);
 	if (m && !m_motor)
 		m_cass->change_state(m_cass->get_state() & CASSETTE_MASK_MOTOR ? CASSETTE_MOTOR_ENABLED : CASSETTE_MOTOR_DISABLED, CASSETTE_MASK_MOTOR);
 	m_motor = m;
+	if (s != m_strobe)
```
Comment (startaq) [inline src/mess/drivers/spc1000.c]: Do you really need to save the old state here? Does m_centronics->write_strobe(!BIT(data, 2)) not work?

File: src/mess/drivers/spc1000.c (hunk 1 above)
Comment (meesokim) [inline src/mess/drivers/spc1000.c]: Yes, you are right. it can be removed.

File: src/mess/drivers/spc1000.c (hunk 2 above)
Comment (meesokim) [inline src/mess/drivers/spc1000.c]: It works fine. we don't need m_strobe variable any more.

File: src/mess/drivers/spc1000.c [hunk 3]
```diff
@@ -217,10 +222,12 @@ READ8_MEMBER(spc1000_state::iplk_r)
 WRITE8_MEMBER( spc1000_state::cass_w )
 {
 	bool m = BIT(data, 1) ? true : false;
+	bool s = ;
```
Comment (meesokim) [inline src/mess/drivers/spc1000.c]: This line should be removed.

**Comment by meesokim** [comment]
I've made another a PR based on startaq's comment. so I'd like to close it.

**Pull request closed without merging**